packages/flutter_tools/lib/src/commands/config.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import '../globals.dart';
+import '../runner/flutter_command.dart';
+
+class ConfigCommand extends FlutterCommand {
+  ConfigCommand() {
+    String usageStatus = flutterUsage.enabled ? 'enabled' : 'disabled';
+
+    argParser.addFlag('analytics',
+      negatable: true,
+      help: 'Enable or disable reporting anonymously tool usage statistics and crash reports ($usageStatus).');
+  }
+
+  @override
+  final String name = 'config';
+
+  @override
+  final String description =
+    'Configure Flutter settings.\n\n'
+    'The Flutter tool anonymously reports feature usage statistics and basic crash reports to help improve\n'
+    'Flutter tools over time. See Google\'s privacy policy: www.google.com/intl/en/policies/privacy.';
+
+  @override
+  final List<String> aliases = <String>['configure'];
+
+  @override
+  bool get requiresProjectRoot => false;
+
+  /// Return `null` to disable tracking of the `config` command.
```

Comment:
presumably some commands have it disabled (because they don't override it)... I think it might be clearer if it was abstract here and we explicitly had to make a choice for each command.